x-pack/plugins/observability_solution/apm/public/utils/get_timestamp.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function getTimestamp(timestamp?: string, fallbackTimestamp?: number): number {
+  if (!fallbackTimestamp && timestamp) {
+    return new Date(timestamp).getTime() * 1000;
```

Comment:
This will truncate the microsecond precision when the timestamp is of type `date_nanos`. Is there a way to retain the microsecond precision?

Reply:
I did a refactor to keep the microsecond precision